relayer/src/foreign_client.rs

```diff
@@ -642,8 +642,20 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
             .consensus_state(client_state.latest_height())?
             .timestamp();
 
+        let current_src_network_time = self
+            .src_chain
+            .query_application_status()
+            .map_err(|e| {
+                ForeignClientError::client_refresh(
+                    self.id().clone(),
+                    "failed querying the application status of source chain".to_string(),
+                    e,
+                )
+            })?
+            .timestamp;
+
         // Compute the duration since the last update of this client
-        let elapsed = Timestamp::now().duration_since(&last_update_time);
+        let elapsed = current_src_network_time.duration_since(&last_update_time);
```

Comment:
I think this change is good. But are we also missing a similar check in `build_update_client_with_trusted()`, specifically for the trusted consensus state? Here we are checking for the latest but we don't check for the trusted we are going to use in the update. 

Maybe we should consider refactoring a bit `build_update_client_with_trusted()`.

Reply:
Let me know what you think Anca!

84a6e23

Reply:
Looks good, added a couple more comments.